temporal-sdk/src/test/java/io/temporal/internal/sync/QueryDispatcherTest.java

```diff
@@ -0,0 +1,44 @@
+/*
+ *  Copyright (C) 2020 Temporal Technologies, Inc. All Rights Reserved.
+ *
+ *  Copyright 2012-2016 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *  Modifications copyright (C) 2017 Uber Technologies, Inc.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"). You may not
+ *  use this file except in compliance with the License. A copy of the License is
+ *  located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ *  or in the "license" file accompanying this file. This file is distributed on
+ *  an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *  express or implied. See the License for the specific language governing
+ *  permissions and limitations under the License.
+ */
+
+package io.temporal.internal.sync;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertThrows;
+
+import io.temporal.common.interceptors.WorkflowOutboundCallsInterceptor;
+import org.junit.Test;
+
+public class QueryDispatcherTest {
+
+  @Test
+  public void testQueryDispatcherException() {
+    QueryDispatcher qd = new QueryDispatcher(null);
+    WorkflowOutboundCallsInterceptor.RegisterQueryInput requestA =
+        new WorkflowOutboundCallsInterceptor.RegisterQueryInput("QueryA", null, null, null);
+    qd.registerQueryHandlers(requestA);
+    Throwable exception =
+        assertThrows(
+            IllegalArgumentException.class,
+            () -> {
+              qd.handleQuery("Query", null);
+            });
+    assertEquals("Unknown query type: Query, knownTypes=[QueryA]", exception.getMessage());
+  }
```

Comment:
Can we also add a positive scenario test? (e.g. when query can be resolved successfully)